src/server.ts

```diff
@@ -249,25 +260,38 @@ export class Server {
    * }
    * ```
    *
-   * For example, you'll want to examine `offerResults` to add affordances
-   * like these to your app:
+   * For example, you'll want to examine `offerResults` to add affordances like
+   * these to your app:
    * * If `wasImmediatelyFilled` is true, then no offer was created. So if you
-   * normally watch the `Server.offers` endpoint for offer updates, you instead
-   * need to check `Server.trades` to find the result of this filled offer.
+   *   normally watch the `Server.offers` endpoint for offer updates, you
+   *   instead need to check `Server.trades` to find the result of this filled
+   *   offer.
    * * If `wasImmediatelyDeleted` is true, then the offer you submitted was
-   * deleted without reaching the orderbook or being matched (possibly because
-   * your amounts were rounded down to zero). So treat the just-submitted offer
-   * request as if it never happened.
-   * * If `wasPartiallyFilled` is true, you can tell the user that `amountBought`
-   * or `amountSold` have already been transferred.
-   *
-   * @see [Post Transaction](https://www.stellar.org/developers/horizon/reference/endpoints/transactions-create.html)
+   *   deleted without reaching the orderbook or being matched (possibly because
+   *   your amounts were rounded down to zero). So treat the just-submitted
+   *   offer request as if it never happened.
+   * * If `wasPartiallyFilled` is true, you can tell the user that
+   *   `amountBought` or `amountSold` have already been transferred.
+   *
+   * @see [Post
+   * Transaction](https://www.stellar.org/developers/horizon/reference/endpoints/transactions-create.html)
    * @param {Transaction} transaction - The transaction to submit.
-   * @returns {Promise} Promise that resolves or rejects with response from horizon.
+   * @param {object} [opts] Options object
+   * @param {boolean} [opts.skipMemoRequiredCheck] - Allow skipping memo
+   * required check, default: `false`. See
+   * [SEP0029](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0029.md).
+   * @returns {Promise} Promise that resolves or rejects with response from
+   * horizon.
    */
   public async submitTransaction(
     transaction: Transaction,
+    opts: Server.SubmitTransactionOptions = { skipMemoRequiredCheck: false },
   ): Promise<Horizon.SubmitTransactionResponse> {
+    // only check for memo required if skipMemoRequiredCheck is false and the transaction doesn't include a memo.
+    if (!opts.skipMemoRequiredCheck && transaction.memo.type === "none") {
+      await this.checkMemoRequired(transaction);
```

Comment:
I'd prefer moving the validation `transaction.memo.type === "none"` inside the `checkMemoRequired` method. Something like:

```ts
public async checkMemoRequired(transaction: Transaction): Promise<void> {
    if (transaction.memo.type !== "none") {
        Promise.resolve();
        return;
    }
    // Continue with validation
}
```

Otherwise you could get the `AccountRequiresMemoError` from calling `checkMemoRequired` in transaction with valid memo type.

Reply:
@marcelosalloum good point!

Reply:
done